core/impl/src/main/java/org/eclipse/sensinact/core/model/nexus/emf/compare/EMFCompareUtil.java

```diff
@@ -122,7 +129,48 @@ private static void serviceUpdate(EReference reference, Provider incomming, Prov
                             + newService.eClass().getName() + " must be a subtype of " + oldService.eClass().getName());
                 }
             }
-            mergeAndNotify(reference, newService, oldService, blackList, accumulator);
+            mergeAndNotify(reference.getName(), newService, oldService, blackList, accumulator);
+        }
+    }
+
+    private static void servicesMapUpdate(DynamicProvider incomming, DynamicProvider original,
+            NotificationAccumulator accumulator,
+            List<EStructuralFeature> blackList) {
+
+        List<String> toDelete = new ArrayList<>(original.getServices().keySet());
```

Comment:
If original is not a `DynamicProvider` then `toDelete` and `toUpdate` are just empty